colcon_devtools/verb/version_check.py

```diff
@@ -31,16 +33,38 @@ def main(self, *, context):  # noqa: D102
         for entry_point in colcon_extension_points.values():
             distributions.add(entry_point.dist)
 
-        pypi = ServerProxy('https://pypi.python.org/pypi')
+        base_url = 'https://pypi.org/pypi/{project}/json'
         for dist in sorted(distributions, key=lambda d: d.project_name):
-            versions = pypi.package_releases(dist.project_name)
-            if not versions:
+            url = base_url.format(project=dist.project_name)
+
+            try:
+                response = urllib.request.urlopen(url)
+            except urllib.error.HTTPError as e:
+                if e.code == 404:
+                    print(
+                        '{dist.project_name}: could not find package on PyPI'
+                        .format_map(locals()), file=sys.stderr)
+                else:
+                    print(
+                        '{dist.project_name}: Server could not fulfill the '
+                        'request: {e.reason}'
+                        .format_map(locals()), file=sys.stderr)
+                continue
+            except urllib.error.URLError:
+                print(
+                    '{dist.project_name}: Failed to reach server'
+                    .format_map(locals()), file=sys.stderr)
+                continue
+
+            try:
```

Comment:
Are there any cases left in which either `response` is `None` or the `status` is not `200` in this line?

Reply:
For `None`, this statement from the urlopen docs leads me to believe that unless the global handler is changed, we shouldn't ever see `None` here:
> Note that `None` may be returned if no handler handles the request (though the default installed global `OpenerDirector` uses `UnknownHandler` to ensure this never happens).

---

For non-200 HTTP responses, it seems that it's up to the HTTP error processor: https://docs.python.org/3/library/urllib.request.html#httperrorprocessor-objects
From what I can tell, the default handler doesn't have any special handling, and all non-successful codes result in an `HTTPError`.

So I'm not 100% sure, but moderately confident that we're reasonably handling all cases here.